### PR TITLE
Geometry tests

### PR DESCRIPTION
**Pull request by Athight**
tests for "closest_within_radius", "intersects" and "closest_non_intersecting_within_radius" in geometry/common.py
also updated some docs to clarify edge cases

### CODE REVIEW

File: tests/test_geometry_common.py [hunk 1]
```diff
@@ -152,5 +153,167 @@ def test_closest_two_equal_targets(self):
...
+        intersects = common.intersects(geom1, geom2)
+        self.assertEqual(intersects, True)
+
+    def test_intersects_point_on_end_of_line_intersects(self):
+        geom1 = Point(0, 0)
+        geom2 = LineString([(0, 0), (2, 2)])
+        intersects = common.intersects(geom1, geom2)
+        self.assertEqual(intersects, True)
+
+    def test_intersects_line_in_linear_ring_does_not_intersect(self):
```
Comment (jeznag) [inline tests/test_geometry_common.py]: I was a bit confused by this one - perhaps `line_inside_linear_ring`

File: tests/test_geometry_common.py [hunk 2]
```diff
@@ -152,5 +153,167 @@ def test_closest_two_equal_targets(self):
...
+        closest_geometry = common.closest_within_radius(geom, targets, 10, 2)
+        self.assertEqual(closest_geometry, Point(10, 0))
+
+    def test_closest_within_radius_second_point_just_outside_radius(self):
+        geom = Point(0, 0)
+        targets = [Point(0, 0), Point(10, 0.000001)]
+        closest_geometry = common.closest_within_radius(geom, targets, 10, 2)
+        self.assertEqual(closest_geometry, None)
+
+# -----------------------------------------------------------------------------
```
Comment (jeznag) [inline tests/test_geometry_common.py]: Could it be worth adding a test for the case where you request point #3 but there are only two points? That case is mentioned in the method description but can't see it directly being tested.

File: tests/test_geometry_common.py [hunk 3]
```diff
@@ -152,5 +153,167 @@ def test_closest_two_equal_targets(self):
...
+
+# closest_non_intersecting_within_radius
+
+    def test_closest_non_intersecting_within_radius_nearest_point_no_intersect(self):
+        geom1 = Point(0, 0)
+        geom2 = Point(1, 2)
+        geomList3 = [Point(1, 1), Point(2, 2)]
+        closest = common.closest_non_intersecting_within_radius(geom1, geom2, geomList3, 10)
```
Comment (jeznag) [inline tests/test_geometry_common.py]: I find it a bit hard to understand method calls like this with long parameter lists where the arguments are quite similar. Might be worth considering using named parameters in the call, e.g.

```
closest = common.closest_non_intersecting_within_radius(geom=geom1, non_intersect_geom=geom2, targets=geomList3, radius=10)
```

Ref https://www.matheus.ro/2018/01/29/clean-code-avoid-many-arguments-functions/

File: tests/test_geometry_common.py [hunk 4]
```diff
@@ -152,5 +153,167 @@ def test_closest_two_equal_targets(self):
...
+        targets = [Point(0, 0), Point(10, 0)]
+        closest_geometry = common.closest_within_radius(geom, targets, 10, 2)
+        self.assertEqual(closest_geometry, Point(10, 0))
+
+    def test_closest_within_radius_second_point_just_outside_radius(self):
+        geom = Point(0, 0)
+        targets = [Point(0, 0), Point(10, 0.000001)]
+        closest_geometry = common.closest_within_radius(geom, targets, 10, 2)
+        self.assertEqual(closest_geometry, None)
+
```
Comment (jeznag) [inline tests/test_geometry_common.py]: Could it be worth adding a test for the case where you request point #3 but there are only two points? That case is mentioned in the method description but can't see it directly being tested.

File: tests/test_geometry_common.py [hunk 5]
```diff
@@ -152,5 +153,167 @@ def test_closest_two_equal_targets(self):
...
+# closest_non_intersecting_within_radius
+
+    def test_closest_non_intersecting_within_radius_nearest_point_no_intersect(self):
+        geom1 = Point(0, 0)
+        geom2 = Point(1, 2)
+        geomList3 = [Point(1, 1), Point(2, 2)]
+        closest = common.closest_non_intersecting_within_radius(geom1, geom2, geomList3, 10)
+        self.assertEqual(closest, Point(1, 1))
+
+    def test_closest_non_intersecting_within_radius_nearest_point_after_intersect(self):
```
Comment (jeznag) [inline tests/test_geometry_common.py]: this is getting quite complicated and hard to put in a method name - perhaps you could put a comment explaining the scenario with an example

File: tests/test_geometry_common.py (hunk 1 above)
Comment (Athight) [reply tests/test_geometry_common.py]: updated to use inside 'instead' of 'in'

File: tests/test_geometry_common.py (hunk 2 above)
Comment (Athight) [reply tests/test_geometry_common.py]: good catch, yes, done

File: tests/test_geometry_common.py (hunk 4 above)
Comment (Athight) [reply tests/test_geometry_common.py]: duplicate comment maybe? done for closest_non_intersecting_within_radius too

File: tests/test_geometry_common.py (hunk 3 above)
Comment (Athight) [reply tests/test_geometry_common.py]: renamed the variables to hopefully fulfil the same effect, happy to use named parameters if this still feels insufficient

File: tests/test_geometry_common.py (hunk 5 above)
Comment (Athight) [reply tests/test_geometry_common.py]: I extracted the method name being tested from the test method and broke it into 4 classes, hopefully that makes it a bit more tolerable 😅 
happy to add describing comments on any tests that still feel too complex

File: tests/test_geometry_common.py [hunk 6]
```diff
@@ -35,285 +34,314 @@ def test_second_closest_point_two_targets(self):
...
+        closest = common.closest_non_intersecting_within_radius(geom, non_intersect_geom, targets, 10)
         self.assertEqual(closest, None)
 
-    def test_closest_non_intersecting_within_radius_nearest_point_closest_counts_through_geom(self):
-        geom1 = Point(0, 0)
-        geom2 = LinearRing([(0.5, 0.5), (0.5, 1.5), (1.5, 1.5), (1.5, 0.5)])
-        geomList3 = [Point(1, 1), Point(2, 2)]
-        closest = common.closest_non_intersecting_within_radius(geom1, geom2, geomList3, 10)
+    #    %%%%%%%   
+    #    %     %
```
Comment (jeznag) [inline tests/test_geometry_common.py]: Nice ascii art!

File: tests/test_geometry_common.py [hunk 7]
```diff
@@ -35,285 +34,314 @@ def test_second_closest_point_two_targets(self):
...
+    # X  %  A  %  B
+    #    %     %   
+    #    %%%%%%%   
+    # consider the above situation:
+    # - given point X as the geometry
+    # - with '%' representing non-intersect geometry (a linear ring in this case)
+    # - which point is closer A or B
+    # if we just care about shortest distance,
+    # then A is the closest
+    # if we want the nearest where we can draw a line that doesn't intersect any of the non-intersect geometry,
```
Comment (jeznag) [inline tests/test_geometry_common.py]: In this case would the line need to curve around?
image

Comment (Athight) [reply tests/test_geometry_common.py]: your interpretation is correct, though currently A is returned as the closest point (this test will serve to check if this changes in future)
